src/myrtlespeech/model/transducer.py

```diff
@@ -0,0 +1,584 @@
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+import torch
+
+
```

Comment:
I agree that it would be good to isolate the `TransducerEncoder` and `TransducerPredNet` from the `Transducer` class by `mv transducer.py/TransducerEncoder -> rnnt.py/RNNTEncoder`. 

Do you mean getting rid of the `TransducerEncoder` etc protos as they stand and replacing with a much higher level one that just takes `hidden_size` (i.e. more similar to the ds1 proto and less like the ds2 proto)? 

I agree that this could be useful longer term when we have converged on a model design but am slightly loathe to do this for the time being as I think the ability to drag and drop optional layers in the protos allows for quick experimentation. 
Another approach that also allows quick experimentation is high-level protos and editing of the `RNNT` class but I find this can be quite painful when performing experiments attached to different commits (e.g. on different gpus on one of the ML machines).

Reply:
OK - agreed on making them specific then.

The protobufs are actually of the antithesis of rapid iteration on the model - modifying the architecture beyond simple size changes requires updating a lot of code/builders etc. The protobufs were really only ever meant to capture known-good configurations with quick experimentation happening elsewhere such as in notebooks where everything else can be built and then the model architecture swapped out for one defined in the notebook via `class MyExperiment(torch.nn.Module):`. They _are_ good at large hyperparameter sweeps for a given architecture but this model is so new that bigger gains are to be had elsewhere. 

Rewriting a caffe-like [prototxt](https://gist.githubusercontent.com/weiliu89/2ed6e13bfd5b57cf81d6/raw/758667b33d1d1ff2ac86b244a662744b7bb48e01/VGG_ILSVRC_16_layers_fc_reduced_deploy.prototxt) frontend should be avoided as much as possible.

The existing ones present here are probably fine for now after the agreed change - better to get something in.

Reply:
Wow ok that prototext frontend is hellish! Ok completely agreed that we should edit this once we have a more stable model. 

Interesting - you're experimenting by defining the model in a notebook - that makes sense - I'll start using that where appropriate.

Reply:
`mv transducer.py/TransducerEncoder -> rnnt.py/RNNTEncoder` completed.